src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Windows.targets

```diff
@@ -105,6 +105,9 @@ The .NET Foundation licenses this file to you under the MIT license.
       <LinkerArg Condition="'$(CETCompat)' == 'false' and '$(_targetArchitecture)' == 'x64'" Include="/CETCOMPAT:NO" />
       <!-- Enable EHCONT if CET is not disabled and CFG is enabled. -->
       <LinkerArg Condition="'$(CETCompat)' != 'false' and '$(_targetArchitecture)' == 'x64' and '$(ControlFlowGuard)' == 'Guard'" Include="/guard:ehcont"/>
+
+      <!-- Export the managed entry point if building a native library in custom main mode -->
```

Comment:
This is not Windows specific logic, so it should be in the non-OS specific Microsoft.NETCore.Native.targets.

Reply:
I wasn't sure whether Xamarin was doing some other stuff it might conflict with. Will move it there 👍

Reply:
We would want to have different defaults between Windows and non-Windows.

@ivanpovazan Does Xamarin iOS depend on the exports file generated by the runtime? It does not look like that it does, just want to double check.

Reply:
> @ivanpovazan Does Xamarin iOS depend on the exports file generated by the runtime? It does not look like that it does, just want to double check.

I don't think it does and fwiw Xamarin always instructs building static libraries when using NativeAOT.